control/summary_table.go

```diff
@@ -16,29 +16,62 @@ type SummaryTable struct {
 }
 
 // NewSummaryTable creates a SummaryTable instance.
-func NewSummaryTable(root xml.Node) SummaryTable {
+func NewSummaryTable(root xml.Node) *SummaryTable {
 	tbl := table{Root: root}
-	return SummaryTable{tbl}
+	return &SummaryTable{tbl}
```

Comment:
My inclination is to stick to passing by value whenever possible...mind trying to change this back, and seeing if it still works?